elasticsearch/values.yaml

```diff
@@ -18,6 +18,24 @@ minimumMasterNodes: 2
 
 esMajorVersion: 6
 
+# Settings for your nodes, clusters, and indices.
+settings:
+  # These settings become part of elasticsearch.yml on the node.
+  # It should be structured, not a text block.
```

Comment:
> to be able to merge these settings

My use case is less about node types and more about different clusters. The scenario I'm envisioning is this:

* There is a baseline configuration (security enabled, https, etc)
* Individual clusters may have slight deviations

For work, we have a single repo housing the configuration for multiple independent clusters, and I want to reduce copy/pasting between clusters. For example, we have a main logs cluster that has a staging implementation which is nearly identical except for sizing (fewer data nodes, smaller disk, etc). What brought me to this merging solution is that I want to explore using SAML, and for that, I want to inherit my base configuration (security enabled, https, change default node_left timeout to 5m, etc) and append SAML settings to it. This cannot be done with extraEnvs (see #53) because it is a list, so either way the configuration (via esConfig or via extraEnvs) would be a full  copy+paste resulting in duplication as well as what I see as some maintenance hazards (diffing similar-purpose dev/staging/prod clusters becomes somewhat annoying, extending the baseline configuration is copy+paste instead of just adding specific additional settings, etc).

Does this make sense?

Reply:
To put a concrete example:

In all my clusters, I'm likely to want the same baseline config (as said, https, security enabled, etc). In the near future, I'm going to explore SAML, and based on my reading so far, each cluster will need it's own SAML ACS per cluster, which means each cluster will need several settings unique per cluster: Under xpack.security.authc.realms.[name], you'd have `sp.entity_id` and `sp.acs` -- different for each cluster. It's possible these could be `cluster` settings shipped with `/_cluster/settings` api, but I haven't tried that.

Roughly, I'm usually looking for ways to reduce maintenance errors and such, and I feel like extending the configuration (vs copy-paste-modify'ing it) is one way to achieve that.